jgrapht-core/src/main/java/org/jgrapht/graph/AsUnweightedDirectedGraph.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * (C) Copyright 2018, by Lukas Harzenetter and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.graph;
+
+import java.io.Serializable;
+
+import org.jgrapht.Graph;
+import org.jgrapht.GraphTests;
+
+/**
+ * Algorithms designed for unweighted directed graphs should also work on weighted directed graphs.
+ * This class emulates an unweighted graph based on a weighted one by returning
+ * <code>Graph.DEFAULT_EDGE_WEIGHT</code> for each edge weight. The underlying weighted graph is
+ * provided at the constructor. Modifying operations (adding/removing vertexes/edges) are also
+ * passed through to the underlying weighted graph. As edge weight, Graph.DEFAULT_EDGE_WEIGHT is
+ * used. Setting an edge weight is not supported. The edges are not modified. So, if an edge is
+ * asked for, the one from the underlying weighted graph is returned. In case the underlying graph
+ * is serializable, this one is serializable, too.
+ *
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ */
+public class AsUnweightedDirectedGraph<V, E>
```

Comment:
This class is not needed at all. In latest versions, we have a dynamic graph type system.